cmd/kubeadm/app/cmd/alpha/selfhosting.go

```diff
@@ -1,161 +0,0 @@
-
-		kubeadm alpha phase self-hosting convert-from-staticpods
-		`)
-)
-
-// newCmdSelfhosting returns the self-hosting Cobra command
-func newCmdSelfhosting(in io.Reader) *cobra.Command {
-	cmd := &cobra.Command{
-		Use:     "selfhosting",
-		Aliases: []string{"selfhosted", "self-hosting"},
-		Short:   "Make a kubeadm cluster self-hosted",
-		Long:    cmdutil.MacroCommandLongDescription,
-	}
-
-	cmd.AddCommand(getSelfhostingSubCommand(in))
-	return cmd
-}
-
-// getSelfhostingSubCommand returns sub commands for Self-hosting phase
-func getSelfhostingSubCommand(in io.Reader) *cobra.Command {
-
-	cfg := &kubeadmapiv1beta2.ClusterConfiguration{}
-	// Default values for the cobra help text
-	kubeadmscheme.Scheme.Default(cfg)
-
-	var cfgPath, featureGatesString, kubeConfigFile string
-	forcePivot, certsInSecrets := false, false
-
-	// Creates the UX Command
-	cmd := &cobra.Command{
-		Use:     "pivot",
-		Aliases: []string{"from-staticpods"},
-		Short:   "Convert a static Pod-hosted control plane into a self-hosted one",
-		Long:    selfhostingLongDesc,
-		Example: selfhostingExample,
-		RunE: func(cmd *cobra.Command, args []string) error {
-
-			var err error
-
-			if !forcePivot {
-				fmt.Println("WARNING: self-hosted clusters are not supported by kubeadm upgrade and by other kubeadm commands!")
-				fmt.Print("[pivot] are you sure you want to proceed? [y/n]: ")
-				s := bufio.NewScanner(in)
-				s.Scan()
-
-				if err = s.Err(); err != nil {
-					return err
-				}
-
-				if strings.ToLower(s.Text()) != "y" {
-					return errors.New("aborted pivot operation")
-				}
-			}
-
-			fmt.Println("[pivot] pivoting cluster to self-hosted")
-
-			if cfg.FeatureGates, err = features.NewFeatureGate(&features.InitFeatureGates, featureGatesString); err != nil {
-				return err
-			}
-
-			if err := validation.ValidateMixedArguments(cmd.Flags()); err != nil {
-				return err
-			}
-
-			// Gets the Kubernetes client
-			kubeConfigFile = cmdutil.GetKubeConfigPath(kubeConfigFile)
-			client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
-			if err != nil {
-				return err
-			}
-
-			// KubernetesVersion is not used, but we set it explicitly to avoid the lookup
-			// of the version from the internet when executing LoadOrDefaultInitConfiguration
-			phases.SetKubernetesVersion(cfg)
-
-			// This call returns the ready-to-use configuration based on the configuration file that might or might not exist and the default cfg populated by flags
-			internalcfg, err := configutil.LoadOrDefaultInitConfiguration(cfgPath, &kubeadmapiv1beta2.InitConfiguration{}, cfg)
-			if err != nil {
-				return err
-			}
-
-			// Converts the Static Pod-hosted control plane into a self-hosted one
-			waiter := apiclient.NewKubeWaiter(client, 2*time.Minute, os.Stdout)
-			return selfhosting.CreateSelfHostedControlPlane(constants.GetStaticPodDirectory(), constants.KubernetesDir, internalcfg, client, waiter, false, certsInSecrets)
```

Comment:
please remove the selfhosting package under phases too:
`k8s.io/kubernetes/cmd/kubeadm/app/phases/selfhosting`

Reply:
fixed